app/socket-apis/upsert-why.js

```diff
@@ -1,16 +1,33 @@
 // https://github.com/EnCiv/civil-pursuit/issues/133
 
-const Points = require('../models/points')
+const Joi = require('joi')
+const { Points, pointSchema } = require('../models/points')
+
+function enforceRequiredFields(schema, requiredFields) { // Add required fields to schema through Joi
+  const schemaDescription = schema.describe()
+  const newSchema = {}
+
+  Object.keys(schemaDescription.keys).forEach((key) => {
+    newSchema[key] = requiredFields.includes(key) ? schema.extract(key).required() : schema.extract(key)
+  })
+
+  return Joi.object(newSchema)
+
+}
 
 async function upsertWhy(pointObj, cb) {
   if (!this.synuser || !this.synuser.id) {
     console.error('upsertWhy called but no user logged in')
     return cb && cb(null) // No user logged in
   }
+
   const userId = this.synuser.id
   pointObj.userId = userId // Add userId to the document
 
-  const validation = Points.validate(pointObj)
+  const requiredFields = ['_id', 'title', 'description', 'userId']
+  const schema = enforceRequiredFields(pointSchema, requiredFields)
+  const validation = schema.validate(pointObj) // Validate pointObj
```

Comment:
This is looking good.  Now lets restructure it a little to make it more reusable.

- [ ] In models/point.js create 

validateRequiredFields(requiredFields,doc)  

where doc is the object to test. 

also make Point.validateRequiredFields work.

then reach socket-api can 
import Point from "../models/point"

and use Point.validateRequiredFields(requiredFields, doc)

it knows validating the point model because that's where it was imported from.

And so it's one line to import, and one line to run in each of the socket-apis.

- [ ] also put enForceRequiredFields into it's own file in app/models/lib  every model that uses it will import it from there.

One question, and you should create a test for this, _id must be an OBJECTID  string coming into the api, and must be an object when written to the database.  

How can we have it both ways? 

Thanks.